composer.json

```diff
@@ -13,14 +13,17 @@
     "minimum-stability": "dev",
     "prefer-stable": true,
     "require": {
-        "php": "^7.1",
-        "amphp/artax": "^3.0",
-        "league/uri-schemes": "^1.0",
+        "php": "^7.2",
+        "amphp/http-client": "^4.1.0",
+        "amphp/http-client-cookies": "^1",
+        "amphp/file": "^1",
+        "league/uri": "^6",
```

Comment:
This would open us up to breaking changes? or?

Reply:
Depends on the public API. Are all classes meant to be public API or just the command line interface?

Reply:
Interesting - it's just command line but we don't commit the lock - it means tests could start failing - I would  stick to stable deps out of habit.

Reply:
What does stable deps mean?

Reply:
So depending on `x.x` instead of `x` (no breaking changes).

Reply:
As long as the package follows semantic versioning, there will be no breaking changes with the way the dependency is defined.

Reply:
I didn't know that. My understanding was that `^6` will allow major versions 7,8,9,etc. :+1:

Reply:
So this is shorthand for `^6.0` basically?

Reply:
Yes, `^` is equal to `>=` + same major version. The other one is `~` I think which works the other way.

Reply:
With that in mind -- any reason to not have `http-client ^4` ?

Reply:
It's using features introduced in 4.1

Reply:
So should it be `^4.1`? or I really didn't understand how the `^` worked :)

Reply:
^4.1 is equivalent to ^4.1.0,yes.